tests/clients/git/test_git.py

```diff
@@ -0,0 +1,168 @@
+import string
+
+from jumpscale.loader import j
+from tests.base_tests import BaseTests
+
+
+class GitTests(BaseTests):
+    def setUp(self):
+        super().setUp()
+        self.instance_name = self.random_name()
+        self.repo_dir = self.random_name()
+        self.repo_url = "https://github.com/tfttesting/test.git"
+        j.sals.fs.mkdir(f"/tmp/{self.repo_dir}")
+        j.sals.process.execute(f"git clone {self.repo_url}", cwd=f"/tmp/{self.repo_dir}")
+        self.git_client = j.clients.git.new(name=self.instance_name, path=f"/tmp/{self.repo_dir}/test/")
+
+    def tearDown(self):
+        j.clients.git.delete(self.instance_name)
+        j.sals.fs.rmtree(path=f"/tmp/{self.repo_dir}")
+
+    def random_name(self):
+        return j.data.idgenerator.nfromchoices(10, string.ascii_letters)
+
+    def test01_check_git_config(self):
+        """Test case for checking git config file.
+
+        **Test Scenario**
+        - Get a git client.
+        - Read the git.config file.
+        - Check that remote_url equal to repo_url.
+        - Check that url equal to remote_url.
+        """
+        self.info("Read the git.config file")
+        git_config = j.sals.fs.read_file(f"/tmp/{self.repo_dir}/test/.git/config")
```

Comment:
I think repo name should be a variable in the setUp method, and used in all tests.